### ISSUE TITLE
The cargo maker doesn't mark warnings as warnings

### ISSUE THREAD

**Issue by MaikKlein** [neomake/neomake#1919]
### Expected behavior

The `cargo` marker doesn't mark warnings as warnings.
![](https://i.imgur.com/LGIHW6D.png)

Because of this `let g:neomake_warning_sign` won't be used.
### Steps to reproduce

[log file](https://gist.github.com/MaikKlein/0e2b3b833f15b8593b9d5f768911d7f8)

#### Neomake debug information

Async support: 1
Current filetype: rust
Windows: 0
[shell, shellcmdflag, shellslash]:['/usr/bin/fish', '-c', 0]

##### Enabled makers

For the current filetype ("rust", used with :Neomake):
 - cargo
   - append_file: 0
   - args: ['check', '--message-format=json', '--quiet']
   - auto_enabled: 1
   - cwd: '/home/maik/projects/sm'
   - exe: 'cargo'
   - process_output: function('neomake#makers#ft#rust#CargoProcessOutput')
NOTE: you can define g:neomake_rust_enabled_makers to configure it (or b:neomake_rust_enabled_makers).

Default maker settings:
 - output_stream: 'both'
 - remove_invalid_entries: 0
 - buffer_output: 1

For the project (used with :Neomake!):
None.
NOTE: you can define g:neomake_enabled_makers to configure it.

###### New-style (dict, overrides old-style)

```
g:neomake: {
  'automake': {
    'ignore_filetypes': ['startify'],
    'events': {
      'BufWritePost': {
        'delay': 0,
      },
    },
  },
}
b:neomake: unset
```

###### Old-style

```
g:neomake_error_sign = {'texthl': 'NeomakeErrorMsg', 'text': '▶'}
g:neomake_highlight_columns = 0
g:neomake_info_sign = {'texthl': 'NeomakeWarningMsg', 'text': 'W'}
g:neomake_logfile = '/tmp/neomake.log'
g:neomake_message_sign = {'texthl': 'NeomakeWarningMsg', 'text': 'W'}
g:neomake_place_signs = 1
g:neomake_warning_sign = {'texthl': 'NeomakeWarningMsg', 'text': 'W'}

```
  makeprg=make

#### :version

```
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.2.2/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

#### :messages

```
"src/lib.rs" 1907L, 63158C
"src/lib.rs" 1907L, 63158C written
```

-->

**Comment by Ploppz** [comment]
Is your problem that marker signs don't show up at all, or that they are displayed as errors? Because the latter is my problem, and I'm wondering whether I should make a separate issue for that.

**Comment by MaikKlein** [comment]
@Ploppz The second because they are not marked as warnings and I think neomake falls back to error. Also I wish I could prioritize errors over warnings.

**Comment by blueyed** [comment]
@Ploppz 
Is it about unused imports for you, too?
What cargo version are you using?

**Comment by Ploppz** [comment]
@blueyed Every warning I have seen lately. When I ran neomake in a project just now, also 'field is never used' and 'struct is never used' (there were no more warnings).

cargo version: `cargo 0.26.0 (41480f5cc 2018-02-26)`.
Active toolchain:
```
stable-x86_64-unknown-linux-gnu (default)
rustc 1.25.0 (84203cac6 2018-03-25)
```

**Comment by TerminalWitchcraft** [comment]
Same issue here. Warnings are displayed as errors

**Comment by YorickPeterse** [comment]
Same problem here: every warning produced by `cargo check` is displayed as an error.  The output of `cargo check --message=format=json` in this case can be found here:  [cargo.txt](https://github.com/neomake/neomake/files/2725300/cargo.txt)

**Comment by Luffbee** [comment]
I found the reason for this problem: `neomake#makers#ft#rust#CargoProcessOutput` use `decoded['message']['code']['code'][0]` as `error.type`.
Here is part of a warning, the `error.type` for this will be 'u':
``` json
{
    "reason": "compiler-message",
    "package_id": "rust-test 0.1.0 (path+file:///Users/luffbee/code/rust-test)",
    "message": {
        "message": "unused `std::result::Result` that must be used",
        "code": {
            "code": "unused_must_use",
            "explanation": null
        },
        "level": "warning",
    }
}

**Comment by blueyed** [comment]
@Luffbee 
Thanks for the info.  Can you provide the full output (for a test), please?
(The previous comment has something in this regard already though, but just for completeness)

**Comment by Luffbee** [comment]
I pasted it here [https://paste.ubuntu.com/p/pKRdt2zZwT/](url). @blueyed

**Comment by blueyed** [comment]
@Luffbee 
Went ahead and merged it - sorry for doing a PR already in parallel, and thanks again for providing a fix yourself.. :v:
